packages/marshal/docs/marshal-vs-store-level.md

```diff
@@ -0,0 +1,95 @@
+
+Each remaining **passable** is either
+  * An *error* carrying diagnostic information.
+  * A *promise*, providing the eventual distributed access that JavaScript's promises were designed to provide.
+  * A *metaTagged*, like the copyTagged, is the extension point used by the Store level to implement
+ *patternNodes*. The distinction is that a metaTagged is necessarily *not* a structure or a key. Thus, a patternNode, encoded into a metaTagged, can be applied to a structure or key specimen, knowing that the specimen cannot itself contain a patternNode. This avoids a tricky source of ambiguity and confusion.
+
+**`sameStructure(s: Structure, t: Structure) => boolean`**<br>
+ implements the most precise equivalence class meaningful at the Marshal layer. However, it is narrow, admitting only structures.
+
+**`comparePassables(left: Passable, right :Passable) => -1 | 0 | 1`**<br>
+implements a full order over *all* passables, suitable for sorting with `Array.prototype.sort`. However, it is a very imprecise full order. For example, for any two remotables `r` and `q`, `comparePassables(r,q) === 0`. All remotables are therefore equivalent in this order. Likewise for all errors, all promises, and all metaTaggeds. Thus, `comparePassable`'s zero answer provides an equivalence class over all passables, but an imprecise one.
+
+ ## The Store level organized by `keyStyleOf`
+
+**`keyStyleOf(p: Passable) => KeyStyle`**<br>
+ where `KeyStyle` is an enumeration quite similar to `PassStyle`
+
+`KeyStyle`:
+   * The Store's **primitive styles** are identical to Marshal's.
+   * The Store's **data styles** styles preserves the `"copyArray"` and `"copyRecord"` classification from Marshal's data styles, omits `"copyTagged"`, but uses it to implement the `"copySet"` and `"copyMap"` that it introduces. The store level must also recognize whether a copyTagged that claims to be a copySet or copyMap has a valid representation. A copyTagged that does not validate is not a key.
+   * Store's **"remotable"** is identical to Marshal's. Every remotable is a key. On two remotables `p` and `q`, `sameKey(p,q)` compares using the remotable's unique unforgeable identity. Thus, for two remotables, `sameKey(p,q)` iff `sameStructure(p,q)` iff `p === q` iff `Object.is(p,q)` iff `sameValueZero(p,q)`.
+   * The remaining passables: errors, promises, and metaTaggeds, are not keys. `keyStyleOf` applied to any of them throws.
+
+Primitives and remotables are keys. copyArrays, copyRecords, copySets, and copyMaps are key-like containers. If a key-like container contains only keys, then it is a key.
+  * A *copySet* can only contain keys, and so each copySet is necessarily a key.
+  * A *copyMap* entry can use only use a key as a key, but can use any passable as a value including unrecognized copyTaggeds. But a copyMap is a key only of all of its values are keys.
```

Comment:
You said above that everything (including `copyMap`) must be hardened to qualify as a passable, and therefore as a key. But that doesn't make the Map contents immutable. Does the `copyMap`'s identity (as a key) remain stable in the face of additions/removals/changes to the `copyMap` contents?

Reply:
Also, does `sameKey` on a `copyMap` require a full examination of the contents every time? I imagine you have some clever trick up your sleeve, but using any container as a key (in a language that lacks hashable object identities) sounds pretty expensive.